src/main/java/com/google/api/codegen/config/DiscoveryMethodModel.java

```diff
@@ -297,11 +297,9 @@ public boolean isIdempotent() {
   public Map<String, String> getResourcePatternNameMap(Map<String, String> nameMap) {
     Map<String, String> resources = new LinkedHashMap<>();
     for (Map.Entry<String, String> entry : nameMap.entrySet()) {
-      String resourceNameString =
-          DiscoGapicNamer.getResourceIdentifier(entry.getKey()).toLowerCamel();
-      if (DiscoGapicNamer.getResourceIdentifier(method.flatPath())
-          .toLowerCamel()
-          .equals(resourceNameString)) {
+      if (DiscoGapicNamer.getCanonicalPath(method).equals(entry.getKey())) {
```

Comment:
wait, I didn't know we were using static methods on DiscoGapicNamer - that prevents language-specific customization, right?

Reply:
The static methods are either
- language-agnostic functions, like in configgen or for parsing the `"list"` method name from the Discovery doc method name `compute.addresses.list`
- functions that return a Name object, so they can be formatted by a language-specific NameFormatter

I should be able to convert all the static methods to non-static methods if you'd think that'd make for better style or testing.

Reply:
I think non-static would be best. In particular for functions returning Name, individual languages might want different name content, not simply different casing.

Reply:
Ok, removed all static methods from DiscoGapicNamer. In addition, removed the SurfaceNamer as a field inside DiscoGapicNamer and made the Surface a parameter in DiscoGapicNamer methods that require it. This allows DiscoGapicNamer to be used to generate language-agnostic config names without having a null inner SurfaceNamer.